### ISSUE TITLE
Unhelpful warnings

### ISSUE THREAD

**Issue by sjakobi** [haskell/haddock#854]
When building the ghc docs with the updates from #853, there are [a lot of warnings](https://github.com/haskell/haddock/files/2088133/haddock-warnings.txt).

There are several problems with them:

* Many ambiguous identifier warnings cannot easily be fixed, at least not until https://github.com/haskell/haddock/issues/667 is addressed. E.g.

    ```
    Warning: 'FunPtr' is ambiguous. It is defined
        * at libraries/base/GHC/Ptr.hs:96:17
         * at libraries/base/GHC/Ptr.hs:96:1
         You may be able to disambiguate the identifier by qualifying it or
         by hiding some imports.
         Defaulting to 'FunPtr' defined at libraries/base/GHC/Ptr.hs:96:1
    ``` 

   I think we should maybe elide these warnings if both identifiers are defined in the same file.

* The warnings don't indicate from which source location they arise, not even from which source file. I think we should at least indicate the file name.

**Comment by sjakobi** [comment]
> I think we should maybe elide these warnings if both identifiers names are defined in the same file.

A better heuristic would be to check whether the two names belong to the same type.

**Comment by sjakobi** [comment]
For out-of-scope-warnings like 

```
Warning: 'throw' is out of scope.
```

we should add the recommendation to qualify the identifier.

**Comment by sjakobi** [comment]
What #863 doesn't address, is the lack of source location info. What we could (at least) do is to output `ifaceOrigFilename` with each warning. Does that seem like a good idea?

**Comment by sjakobi** [comment]
Alternatively we could follow the example of the "could not find link destinations" warnings and output

```
Warning: <module name>: <text>
```
